First bunch of fixes

Nakeib opened the pull request:
RC-24: Fix dragging item across window bottom border issue
RC-20: Fix logout button from player inventory window
RC-11: Fix for context menu separator width
RC-12: Fix hotkey input not working when hotkey option window closed
RC-22: Add Tab input key binding to switch channels

Nakeib committed:
RC-24: Fix dragging item across window bottom border issue

Added mouse button pressed condition check to
SidebarWidgetBottomResizer.OnPointerExit event, to determine
if mouse cursor should be changed to default.

Signed-off-by: Adam Czupryna <czupryna.adam@gmail.com>

Nakeib committed:
RC-20: Fix logout button from player inventory window

Added ToggleButtonsWidget.OnLogoutButtonClick implementation
to perform disconnect from server operation.

Signed-off-by: Adam Czupryna <czupryna.adam@gmail.com>

seenquev commented on OpenTibia/Assets/Scripts/Core/Components/SidebarWidgetBottomResizer.cs:
```diff
@@ -91,7 +91,7 @@ public class SidebarWidgetBottomResizer : BasicElement,
 
             // if no drag is occuring and cursor has been pushed, then pop it
-            if (!_mouseHeldDown && _cursorPushed)
+            if (!_mouseHeldDown && !OpenTibiaUnity.InputHandler.IsAnyMousePressed() && _cursorPushed)
```
Looks good assuming there is no way any of these: `OpenTibiaUnity.InputHandler.IsAnyMousePressed()` can return null :)

Nakeib replied:
That's right. There is no way one of those could return null.

Nakeib committed:
RC-11: Fix for context menu separator width

Changed context menu separator width.

Signed-off-by: Adam Czupryna <czupryna.adam@gmail.com>

Nakeib committed:
RC-12: Fix hotkey input not working when hotkey option window closed.

Moved AddKeyUpListener to Awake function body and
RevokeKeyUpListener to OnDestroy function body of
LegacyHotkeyOptionsWidget class.

Signed-off-by: Adam Czupryna <czupryna.adam@gmail.com>

seenquev commented on OpenTibia/Assets/Scripts/UI/Legacy/ContextMenuBase.cs:
```diff
@@ -62,7 +62,7 @@ public class ContextMenuBase : Core.Components.Base.AbstractComponent
             var separatorGO = Instantiate(OpenTibiaUnity.GameManager.HorizontalSeparator, transform);
             var layoutElement = separatorGO.AddComponent<UnityUI.LayoutElement>();
-            layoutElement.minWidth = 120;
+            layoutElement.minWidth = 80;
```
Is that all what it takes? It's no dynamic in any way? I'm still thinking about mobile platforms and supposedly we will have to deal with different resolutions, but let's leave it as it is for now.

Nakeib replied:
Actually this is only minimum separator width. It could be set to 0 as well as later it's fit to the context menu width.

seenquev replied:
So why don't we set it to 0? :)

Nakeib replied:
That was suggestion from more experienced Unity users. :)

seenquev replied:
Let's leave it as it is now then. I'm just afraid it will result in regression in some other part of the system, as we don't really know what is the purpose of having this minimum value set to value > 0.

seenquev commented on OpenTibia/Assets/Scripts/Modules/Options/LegacyHotkeyOptionsWidget.cs:
```diff
@@ -121,10 +121,6 @@ public class LegacyHotkeyOptionsWidget : UI.Legacy.PopUpBase
 
         protected override void OnDisable() {
             base.OnDisable();
-
-            var inputHandler = OpenTibiaUnity.InputHandler;
-            if (inputHandler != null)
-                inputHandler.RemoveKeyUpListener(OnKeyUp);
```
Are we completely sure this listener is removed in some point? There is no way it will persist registered despite of the fact there is no need for that?

Nakeib replied:
I'm pretty sure it's removed on inputHandler destroy. But I am not C# expert, I am not able yet to verify it by just looking into code.

Nakeib replied:
As you can see I've moved adding key up listener to Awake function and revoking this particular listener to OnDestroy function. I've checked each function is called once - at the game start and stop respectively.

Nakeib committed:
RC-22: Add Tab input key binding to switch channels

Added Tab key binding to chat on and chat off default mappings.
This input key performs ChatNextChannel action.

Signed-off-by: Adam Czupryna <czupryna.adam@gmail.com>

seenquev commented on OpenTibia/Assets/Scripts/Modules/ToggleButtons/ToggleButtonsWidget.cs:
```diff
@@ -161,7 +161,27 @@ public class ToggleButtonsWidget : UI.Legacy.SidebarWidget
+            {
+                void OnLogoutPopupYesClick()
+                {
+                    if (gameManager.IsGameRunning)
```
This is redundant, I think - the same check is performed on the line 166

Nakeib replied:
You are right.

seenquev commented on OpenTibia/Assets/Scripts/Modules/ToggleButtons/ToggleButtonsWidget.cs:
```diff
@@ -161,7 +161,27 @@ public class ToggleButtonsWidget : UI.Legacy.SidebarWidget
+                }
 
+                void OnLogoutPopupNoClick()
+                {
```
Shouldn't we somhow log it at least?

Nakeib replied:
I've based on similar parts from existing code. What kind of logging you would like to have?

seenquev replied:
Not pretty sure how it works with Unity, but it'd be nice to have some kind of 'verbose' / 'debug' version of the client in which we could trace each action taken by the user. Testers would be then to drop verbose logs and attach it to every bug they report. In this way we'd have clear view on what was happening with the app at a given time. Is there anything like that in the client?

Nakeib replied:
Yes, there is such mechanism like Debug.Log, .LogWarning, .LogError. But in this case is it worth to log "No" button click from that popup window?

seenquev replied:
Let's then agree on at least put: // Intentionally do nothing here :)

Nakeib replied:
I agree :)

seenquev commented:
LGTM!
